csharp/BYOAuthFlow/Program.cs

```diff
@@ -0,0 +1,228 @@
+            }
+            finally
+            {
+                Cleanup(dsClient, deployment);
+            }
+        }
+
+        private static string CreatePIT(PlayerIdentityTokenServiceClient pitClient)
+        {
+            Console.WriteLine("Generate a PIT");
+            var playerIdentityTokenResponse = pitClient.CreatePlayerIdentityToken(
+                new CreatePlayerIdentityTokenRequest
+                {
+                    Provider = "provider",
+                    PlayerIdentifier = "player_identifier",
+                    ProjectName = ProjectName,
+                });
+            return playerIdentityTokenResponse.PlayerIdentityToken;
+        }
+
+        private static Deployment FindDeployment(DeploymentServiceClient dsc, string deploymentId)
+        {
+            Console.WriteLine("Choosing deployment");
+            return dsc.ListDeployments(new ListDeploymentsRequest
+            {
+                ProjectName = ProjectName
+            }).First(d => d.Id == deploymentId);
+        }
+
+        private static string CreateLoginTokenForDeployment(LoginTokenServiceClient ltClient,
+            string pit, string deploymentId)
+        {
+            Console.WriteLine("Generate a Login Token for the selected deployment");
+            var createLoginTokenResponse = ltClient.CreateLoginToken(
+                new CreateLoginTokenRequest
+                {
+                    PlayerIdentityToken = pit,
+                    DeploymentId = deploymentId,
+                    LifetimeDuration = Duration.FromTimeSpan(new TimeSpan(0, 0, 30, 0)),
+                    WorkerType = "UnityClient",
+                });
+            var loginToken = createLoginTokenResponse.LoginToken;
+            return loginToken;
+        }
+
+        private static void ConnectToLocator(string locatorHost, ushort locatorPort, bool insecureConnection, string pit, string loginToken)
+        {
+            Console.WriteLine("Connect to the deployment using the Login Token and PIT");
+            var locatorParameters = new LocatorParameters
+            {
+                PlayerIdentity = new PlayerIdentityCredentials
+                {
+                    PlayerIdentityToken = pit,
+                    LoginToken = loginToken,
+                },
+              UseInsecureConnection = insecureConnection,
+            };
+
+            var locator = new Locator(locatorHost, locatorPort, locatorParameters);
+            var connectionParameters = CreateConnectionParameters();
+            using (var connectionFuture = locator.ConnectAsync(connectionParameters))
+            {
+                var connectionOption = connectionFuture.Get(5000 /* Using milliseconds */);
+                if (connectionOption.HasValue)
+                {
+                    var connection = connectionOption.Value;
+                    Console.WriteLine("connected: {0}", connection.IsConnected);
```

Comment:
Should "connected" have a capital, for consistency with the other bits of CLI output?